### PR TITLE
Bombae v1.1 : Handle SMS permission like a Marshmallow app

### PR DESCRIPTION
**Pull request by msfjarvis**

### CODE REVIEW

File: app/src/main/java/regalstreak/me/bombae/bombae.java [hunk 1]
```diff
@@ -40,16 +53,16 @@ public void checkEmpty(View view){
         // Start by getting the mobile number.
         number = mobilenotxt.getText().toString();
 
-        if(number != null && !number.isEmpty()){
+        if(!number.equals("") && !number.isEmpty()){
```
Comment (regalstreak) [inline app/src/main/java/regalstreak/me/bombae/bombae.java:56]: what is the use of this?

File: app/src/main/java/regalstreak/me/bombae/bombae.java [hunk 2]
```diff
@@ -40,16 +53,16 @@ public void checkEmpty(View view){
...
 
             // Number wasn't empty. Get the text message to send.
             text = msgtxt.getText().toString();
 
-            if(text != null && !text.isEmpty()){
+            if(!text.equals("") && !text.isEmpty()){
```
Comment (regalstreak) [inline app/src/main/java/regalstreak/me/bombae/bombae.java:61]: this

File: app/src/main/java/regalstreak/me/bombae/bombae.java [hunk 3]
```diff
@@ -40,16 +53,16 @@ public void checkEmpty(View view){
...
                 // Text wasn't empty. Get the number of times.
                 times = timestxt.getText().toString();
 
-                if(times != null && !times.isEmpty()){
+                if(!times.equals("") && !times.isEmpty()){
```
Comment (regalstreak) [inline app/src/main/java/regalstreak/me/bombae/bombae.java:65]: and this?

**Comment by regalstreak** [comment]
see comments on the commit

**Comment by msfjarvis** [comment]
This is the recommended string comparison method to check for a blank string

**Comment by msfjarvis** [comment]
And I thought I'd mention that getText will never be null unless the EditText reference is null, and in that case getText will throw an NPE and crash

**Comment by regalstreak** [comment]
the crash has been handled. Showing a silly snackbar.

**Pull request closed without merging**